File: 프로그래머스/lv0/120842. 2차원으로 만들기/2차원으로 만들기.cpp
#include <string>
#include <vector>

using namespace std;

vector<vector<int>> solution(vector<int> num_list, int n)
{
    vector<vector<int>> v(num_list.size()/n, vector<int>(n, 0));


    int z = 0;
    for (int i = 0; i < num_list.size() / n; i++)
    {
        for (int j = 0; j < n; j++)
        {
            v[i][j] = num_list[z];
            z++;
        }   
    }

    return v;
}

File: 프로그래머스/0/120842. 2차원으로 만들기/2차원으로 만들기.java
class Solution {
    public int[][] solution(int[] num_list, int n) {
        int[][] answer = new int[num_list.length / n][n];
        
        int rowIdx = 0;
        int colIdx = 0;

        int idx = 0;
        while (idx < num_list.length) {
            answer[rowIdx][colIdx++] = num_list[idx++];
            
            if (colIdx == n) {
                rowIdx++;
                colIdx = 0;
            }
        }
        
        return answer;
    }
}